Sources/FoundationEssentials/FileManager/FileManager+Directories.swift

```diff
@@ -149,6 +153,28 @@ extension _FileManagerImpl {
     }
     
     func subpathsOfDirectory(atPath path: String) throws -> [String] {
+#if os(Windows)
+        var results: [String] = []
+        let iterator = _Win32DirectoryContentsSequence(path: path, appendSlashForDirectory: true).makeIterator()
+        while let item = iterator.next() {
+            results.append(item.fileName)
+            if item.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY == FILE_ATTRIBUTE_DIRECTORY &&
+                item.dwFileAttributes & FILE_ATTRIBUTE_REPARSE_POINT != FILE_ATTRIBUTE_REPARSE_POINT {
+
+                var pwszSubPath: PWSTR? = nil
+                let hr = PathAllocCombine(path, item.fileName, PATHCCH_ALLOW_LONG_PATHS, &pwszSubPath)
+                guard hr == S_OK else {
+                    throw CocoaError.errorWithFilePath(path, win32: WIN32_FROM_HRESULT(hr), reading: true)
+                }
+                results.append(contentsOf: try subpathsOfDirectory(atPath: item.fileName).map {
+                    var pwszFullPath: PWSTR? = nil
+                    _ = PathAllocCombine(item.fileName, $0, PATHCCH_ALLOW_LONG_PATHS, &pwszFullPath)
+                    return String(decodingCString: pwszFullPath!, as: UTF16.self).standardizingPath
+                 })
+            }
+        }
+        return results
```

Comment:
Similar comment as below, is there any way for us to implement a `.subpaths` var on `_Win32DirectoryContentsSequence` so that the implementation of dealing with the subpaths themselves could be a single implementation?

Reply:
I could of course move the iteration into a `subpaths` property and then use that to "simplify" `subpathsOfDirectory(atPath:)` to `_Win32DirectoryContentsSequence(...).subpaths`. But that would then also suggest that we should sink the non-Windows path into `_FTSSquence` no?

Reply:
`_FTSSequence` has a `subpaths` property that returns a `Collection<String>` which we then apply logic to here (for surfacing errors and trimming the prefix). I was wondering if it was possible to implement a similar `subpaths` property on `_Win32DirectoryContentsSequence` so that the logic below (lines 190-210) could be shared between the two implementations to ensure equivalent behavior. Is that possible? Or are the APIs too different to do this

Reply:
Oh, the APIs are completely different. It isn't a full enumerated sequence, instead we have to manually do the BFS to get the tree.

Reply:
The non-`FOUNDATION_FRAMEWORK` path below also takes a BFS-style approach since we don't have a full enumeration there either - is that one compatible enough to adapt here?

Reply:
Ah I see now gotcha yeah the semantics are a bit different then. Ok no worries